### ISSUE TITLE
Multiple simulator instances for the same caps can be created with noReset caps

### ISSUE THREAD

**Issue by NozomiIto** [Magic-Pod/AppiumRegressionCheck#14]
How to reproduce:

1. Remove `iPhone 7` x `iOS 10.3.*` simulator instances from the simulator list, if you have them.

<img width="826" alt="screen shot 2018-05-18 at 8 34 14" src="https://user-images.githubusercontent.com/4276719/40208759-45ea8a26-5a76-11e8-9f15-213009245ccc.png">

2. Temporarily fix [iOS1 10 test](https://github.com/Magic-Pod/AppiumRegressionCheck/blob/master/test/testAppium.js#L348) like the following.

<img width="1210" alt="screen shot 2018-05-18 at 8 36 16" src="https://user-images.githubusercontent.com/4276719/40208814-8f189ed6-5a76-11e8-8073-6a44ec6cdcd7.png">

3. Run `npm run test -- --grep "should work with iOS simulator 10"` with appium@1.8.0 and XCode 9.3.1.
4. Then 4 simulator instances for `iPhone7` x `iOS 10.3.1` will be created. They are created for evey test method and continues to increase.

<img width="1144" alt="screen shot 2018-05-18 at 8 28 27" src="https://user-images.githubusercontent.com/4276719/40208859-e24ebbc6-5a76-11e8-9153-2e922fc11d38.png">

Server log for the test run above: https://gist.githubusercontent.com/NozomiIto/d4fd74535128afdb080a8abb34b95488/raw/07a3fc681f6eb935bf32851fafee44ade6803c37/server%2520log

When a user tried to test with iOS version for which no simulator instance existed, this problem happened.

Pull request for https://github.com/appium/appium/issues/9134 seems to try to fix this problem once, but currently it happens again..

**Comment by NozomiIto** [comment]
https://github.com/appium/appium/issues/9134#issuecomment-326383676 says `there is no reason not to use a device, that already exists in the system, considering that device creation and deletion requires additional time.` , so I think this problem is worth being fixed.

**Comment by KazuCocoa** [comment]
![image](https://user-images.githubusercontent.com/5511591/40209414-f4b15a00-5a79-11e8-969b-ab0682a0bfc2.png)

Ah... it looks device names are unique... So, it never matched in https://github.com/appium/appium-xcuitest-driver/blob/master/lib/simulator-management.js#L18

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-xcuitest-driver/blame/master/lib/simulator-management.js#L9
The name of created simulator was implemented for a long time ago.

**Comment by KazuCocoa** [comment]
I read the code.

1. Simulators should create multiple instances.
    - Different identifiers, the same device name is allowed
2. Instances are generated with `appiumTest-${sessionId}`
    - So, if the `sessionId` is re-generated, the instance name will change

What about set `udid` after the first running and use the `udid` in second runing time?

you can see the udid as the response of `createSession` or `@driver.session_capabilities`.

- createSession
```
[debug] [MJSONWP] Responding to client with driver.createSession() result: {"capabilities":{"webStorageEnabled":false,"locationContextEnabled":false,"browserName":"","platform":"MAC","javascriptEnabled":true,"databaseEnabled":false,"takesScreenshot":true,"networkConnectionEnabled":false,"platformName":"ios","automationName":"XCUITest","app":"/Users/kazuaki/GitHub/ruby_lib_core/test/functional/app/UICatalog.app","platformVersion":"10.3","deviceName":"iPhone Simulator","useNewWDA":true,"useJSONSource":true,"someCapability":"some_capability","newCommandTimeout":120,"udid":"DED4DBAD-8E5E-4AD6-BDC4-E75CF9AD84D8"}}
```

- session_capabilities
```
[debug] [JSONWP Proxy] Proxying [GET /] to [GET http://localhost:8100/session/AB69E24A-F0CE-4654-A0E2-FEAA8C371D54] with no body
[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"sessionId\" : \"AB69E24A-F0CE-4654-A0E2-FEAA8C371D54\",\n    \"capabilities\" : {\n      \"device\" : \"iphone\",\n      \"browserName\" : \"UICatalog\",\n      \"sdkVersion\" : \"10.3.1\",\n      \"CFBundleIdentifier\" : \"com.example.apple-samplecode.UICatalog\"\n    }\n  },\n  \"sessionId\" : \"AB69E24A-F0CE-4654-A0E2-FEAA8C371D54\",\n  \"status\" : 0\n}"
[XCUITest] Merging WDA caps over Appium caps for session detail response
[debug] [JSONWP Proxy] Proxying [GET /wda/screen] to [GET http://localhost:8100/session/AB69E24A-F0CE-4654-A0E2-FEAA8C371D54/wda/screen] with no body
[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"statusBarSize\" : {\n      \"width\" : 375,\n      \"height\" : 23.4375\n    },\n    \"scale\" : 2\n  },\n  \"sessionId\" : \"AB69E24A-F0CE-4654-A0E2-FEAA8C371D54\",\n  \"status\" : 0\n}"
[debug] [JSONWP Proxy] Proxying [GET /wda/screen] to [GET http://localhost:8100/session/AB69E24A-F0CE-4654-A0E2-FEAA8C371D54/wda/screen] with no body
[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"statusBarSize\" : {\n      \"width\" : 375,\n      \"height\" : 23.4375\n    },\n    \"scale\" : 2\n  },\n  \"sessionId\" : \"AB69E24A-F0CE-4654-A0E2-FEAA8C371D54\",\n  \"status\" : 0\n}"
[debug] [JSONWP Proxy] Proxying [GET /wda/screen] to [GET http://localhost:8100/session/AB69E24A-F0CE-4654-A0E2-FEAA8C371D54/wda/screen] with no body
[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"statusBarSize\" : {\n      \"width\" : 375,\n      \"height\" : 23.4375\n    },\n    \"scale\" : 2\n  },\n  \"sessionId\" : \"AB69E24A-F0CE-4654-A0E2-FEAA8C371D54\",\n  \"status\" : 0\n}"
[debug] [JSONWP Proxy] Proxying [GET /window/size] to [GET http://localhost:8100/session/AB69E24A-F0CE-4654-A0E2-FEAA8C371D54/window/size] with no body
[debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"width\" : 375,\n    \"height\" : 667\n  },\n  \"sessionId\" : \"AB69E24A-F0CE-4654-A0E2-FEAA8C371D54\",\n  \"status\" : 0\n}"
[debug] [W3C] Responding to client with driver.getSession() result: {"udid":"DED4DBAD-8E5E-4AD6-BDC4-E75CF9AD84D8","platformName":"ios","automationName":"XCUITest","app":"/Users/kazuaki/GitHub/ruby_lib_core/test/functional/app/UICatalog.app","platformVersion":"10.3","deviceName":"iPhone Simulator","useNewWDA":true,"useJSONSource":true,"someCapability":"some_capability","newCommandTimeout":120,"device":"iphone","browserName":"UICatalog","sdkVersion":"10.3.1","CFBundleIdentifier":"com.example.apple-samplecode.UICatalog","pixelRatio":2,"statBarHeight":23.4375,"viewportRect":{"left":0,"top":47,"width":750,"height":1287}}
[HTTP] <-- GET /wd/hub/session/9a10ebdd-25ba-482f-a4dc-10916e4601a8 200 833 ms - 566
[HTTP]
```

example.
https://github.com/appium/appium-xcuitest-driver/blob/ad6ecafbbfe357c29c8583cc0396aa9a87500a39/test/functional/basic/basic-e2e-specs.js#L53

**Comment by KazuCocoa** [comment]
@NozomiIto

**Comment by KazuCocoa** [comment]
![image](https://user-images.githubusercontent.com/5511591/40268608-b4a257ba-5bab-11e8-8635-bcebc13240f0.png)

**Comment by KazuCocoa** [comment]
I'll suggest creating non-session id name.
e.g. `appiumTest-iPhone 7` in this case

**Comment by NozomiIto** [comment]
@KazuCocoa 

Thanks for checking.

> Thus, what about set udid after the first running and use the udid in second runing time?

It is not impossible, but, Magic Pod must remember the all {udid -> OS version x device type} maps even after Appium server is closed (since iOS simulators are not killed even after the Appium server has been closed) and the complicated logic would be required..

So I want to wait your PR for appium-xcuitest-driver to be merged!

**Comment by KazuCocoa** [comment]
The PR has been merged. Will release next time. @NozomiIto

**Comment by NozomiIto** [comment]
I confirm it is fixed with 1.8.1, thanks!
